File: benchmarks/pandas/bench_shift_diff.py
"""Benchmark: Series.shift and Series.diff on 100k-element Series"""
import json, time
import numpy as np
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10

data = np.arange(ROWS, dtype=float) * 1.5
s = pd.Series(data)

for _ in range(WARMUP):
    s.shift(1)
    s.diff(1)

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.shift(1)
    s.diff(1)
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "shift_diff",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_shift_diff.ts
/**
 * Benchmark: shiftSeries and diffSeries on 100k-element Series
 */
import { Series, shiftSeries, diffSeries } from "../../src/index.ts";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

const data = Array.from({ length: ROWS }, (_, i) => i * 1.5);
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  shiftSeries(s, 1);
  diffSeries(s, 1);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  shiftSeries(s, 1);
  diffSeries(s, 1);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "shift_diff",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
